appformer-js-bridge/src/main/java/org/uberfire/jsbridge/client/loading/JsWorkbenchLazyPerspectiveActivity.java

```diff
@@ -0,0 +1,215 @@
+                                                                            isDefault(),
+                                                                            jsPerspective,
+                                                                            placeManager);
+        } else {
+            backedPerspective = new JsWorkbenchPerspectiveActivity(jsPerspective,
+                                                                   placeManager,
+                                                                   isDefault());
+        }
+
+        if (activityManager.isStarted(this)) {
+            // current activity is started, need to move the backed perspective to started state
+            getBackedPerspective().onStartup(place);
+        }
+
+        if (open) {
+            // lazy perspective is opened, need to move the backed perspective to open state and refresh the page
+            getBackedPerspective().onOpen();
+            placeManager.goTo(new ForcedPlaceRequest(backedPerspectiveId));
+        }
+    }
+
+    @Override
+    public PerspectiveActivity get() {
+        if (isPerspectiveLoaded()) {
+            return getBackedPerspective();
+        }
+        return this;
+    }
+
+    // Lifecycle
+
+    @Override
+    public void onStartup(final PlaceRequest place) {
+
+        this.place = place;
+
+        if (isPerspectiveLoaded()) {
+            getBackedPerspective().onStartup(place);
+            return;
+        }
+        super.onStartup(place);
+    }
+
+    @Override
+    public void onOpen() {
+
+        if (isPerspectiveLoaded()) {
+            getBackedPerspective().onOpen();
+        } else {
+            super.onOpen();
+            onLoaded(); // trigger backed perspective loading
+        }
+
+        placeManager.executeOnOpenCallbacks(place);
+    }
+
+    @Override
+    public void onClose() {
+
+        if (isPerspectiveLoaded()) {
+            getBackedPerspective().onClose();
+        } else {
+            super.onClose();
+        }
+
+        placeManager.executeOnCloseCallbacks(place);
+    }
+
+    @Override
+    public void onShutdown() {
+
+        if (isPerspectiveLoaded()) {
+            getBackedPerspective().onShutdown();
+            return;
+        }
+        super.onShutdown();
+    }
+
+    void onLoaded() {
+        lazyLoadingParentScript.accept(backedPerspectiveId);
+    }
+
+    // Properties
+
+    PerspectiveActivity getBackedPerspective() {
+        return backedPerspective;
+    }
+
+    @Override
+    public ResourceType getResourceType() {
+        return ActivityResourceType.PERSPECTIVE;
+    }
+
+    @Override
+    public String getIdentifier() {
+        return backedPerspectiveId;
+    }
+
+    @Override
+    public boolean isDefault() {
+        // we ignore the isDefault() property of the backed perspective,
+        // it shouldn't be different than the one configured for the lazy one
```

Comment:
How do you mean?

Reply:
if the backed perspective's `isDefault()` property is really the same as in the lazy one

Reply:
Yeah.. I think this comment is a little bit misleading. We chose to use the configured property on the lazy one, so there's no need to "test" if they are the same.. we simply ignore the isDefault() property of the backed perspective. I'll remove the second comment line so that's more clear.